tsuba: modified most functions to use KATANA_CHECKED

pyrito opened the pull request:
Now that I have some time, I decided to do some code cleanup in our storage code to make use of `KATANA_CHECKED` wherever I saw it was most appropriate.

Commit:
modified most functions to use KATANA_CHECKED

tylershunt commented on libtsuba/src/AddProperties.cpp:
```diff
@@ -20,18 +20,11 @@ katana::Result<std::shared_ptr<arrow::Table>>
-    return out_res.error().WithContext("loading property");
-  }
+  auto reader =
+      KATANA_CHECKED_CONTEXT(tsuba::ParquetReader::Make(), "loading property");
```
We might take this opportunity to reduce stuttering in errors. I think a good rule of thumb is not to report context that the caller knows. In this case I don't think "loading property" is adding anything that the caller of this doesn't know.

e-mcginnis replied:
+1 
Though personally I do prefer we lean towards over-providing context rather than under-providing it.

pyrito replied:
Agreed. There are certain places I removed it where it didn't provide much value, but in general I think adding some context can be helpful when looking at an error message. Unless @tylershunt you think it's confusing and should only be there if it provides needed context? E.g. if it's just a isolated function call, there may not be any context needed since the error gives the line number?

witchel replied:
I think `KATANA_CHECKED` is easier to read and should be preferred.  `KATANA_CHECKED_CONTEXT` is for cases where the context, beyond file and line number, is valuable and (always?) that includes data values.

tylershunt commented on libtsuba/src/AddProperties.cpp:
```diff
@@ -186,22 +179,16 @@ tsuba::AddPropertySlice(
-            "adding {}", std::quoted(prop->name()));
-      }
+      KATANA_CHECKED_CONTEXT(
+          add_fn(props), "adding: {}", std::quoted(prop->name()));
```
This is an example of good context IMO the caller doesn't know which property triggered the error.

e-mcginnis commented on libtsuba/src/RDG.cpp:
```diff
@@ -613,29 +590,16 @@ tsuba::RDG::Store(
-  }
   // All write buffers must outlive desc
-  std::unique_ptr<WriteGroup> desc = std::move(desc_res.value());
+  auto desc = KATANA_CHECKED(WriteGroup::Make());
```
nit:
I'm sure this is up for debate, but I prefer the explicit `std::unique_ptr<WriteGroup>` here vs the `auto`. I don't like having to look at the signature of `WriteGroup::Make()` or whatever we are calling to know what `desc` is. Especially since its not just a `WriteGroup` like one might assume, but a  `std::unique_ptr<WriteGroup>`.

pyrito replied:
@witchel wanted to get your opinion on this. I'm a fan of using `auto` for most things since it's usually cleaner to write, but I can also see Evan's point here.

witchel replied:
This is an ongoing debate, but I think the non-auto camp is winning at Katana.  I see the value of explicit types for people less familiar with the code.  We should prioritize the needs of this audience.

pyrito replied:
Makes sense. I will make the changes accordingly

e-mcginnis commented on libtsuba/src/RDGPartHeader.cpp:
```diff
@@ -115,21 +112,14 @@ katana::Result<void>
-  }
-
-  std::string serialized = std::move(serialized_res.value());
+  auto serialized = KATANA_CHECKED(katana::JsonDump(*this));
```
nit: same auto nit as above

scober reviewed (approved):
Thank you for doing this

Commit:
changes based on comments

pyrito commented:
I made some updates and addressed some concerns from the reviewers. I left some context prints in areas where I thought it might be helpful but please feel free to disagree.

e-mcginnis reviewed (approved):
Looks great, thanks for cleaning this up!
